src/libsyntax/parse/parser/expr.rs

```diff
@@ -1084,7 +1084,10 @@ impl<'a> Parser<'a> {
                 if let token::Literal(token::Lit { kind: token::Integer, symbol, suffix })
                         = next_token.kind {
                     if self.token.span.hi() == next_token.span.lo() {
-                        let s = String::from("0.") + &symbol.as_str();
+                        // FIXME: Should just be `&symbol.as_str()` but can't as of now due to
+                        // Deref coercion.
+                        // Issue: https://github.com/rust-lang/rust/issues/51916
+                        let s = String::from("0.") + &symbol.as_str().get_str();
```

Comment:
You can write `&*symbol.as_str()` to explicitly deref if I'm not mistaken. Then you don't need the `get_str` method. And in that case, I would probably also remove the `FIXME` comment.

Reply:
Alright, this worked!